library/rsa.c

```diff
@@ -398,23 +398,24 @@ int mbedtls_rsa_export_raw( const mbedtls_rsa_context *ctx,
     return( ret );
 }
 
+int mbedtls_rsa_is_private( const mbedtls_rsa_context *ctx )
+{
+    return(
+        mbedtls_mpi_cmp_int( &ctx->N, 0 ) != 0 &&
```

Comment:
This function returns 0 on an incomplete key. I find this behavior surprising. I'd prefer `mbedtls_rsa_is_private(rsa)` to return the same thing before and after `mbedtls_rsa_complete(rsa)`. If it doesn't, this should be documented clearly.

Reply:
Good point, I agree this should be documented. I'm likely to rework the PR in a way that no longer relies on this function, but I'll make sure to properly document that aspect of the new function.